src/profile-logic/cpu.js

```diff
@@ -49,6 +48,51 @@ export function computeMaxThreadCPUDelta(
   return maxThreadCPUDelta;
 }
 
+/**
+ * Returns the expected cpu delta per sample if cpu is at 100% and
+ * sampling happens at the declared interval.
+ *
+ * Returns null if the profile does not use cpu deltas.
+ * Otherwise, returns a ratio that can be used to compare activity
+ * between threads with cpu deltas and threads without cpu deltas.
+ *
+ * Examples:
+ *  - interval: 2 (ms), sampleUnits: undefined
+ *    Returns null.
+ *  - interval: 5 (ms), sampleUnits.threadCPUDelta: "µ"
+ *    Returns 5000, i.e. "5000µs cpu delta per sample if each sample ticks at
+ *    the declared 5ms interval and the CPU usage is at 100%".
+ *  - interval: 3 (ms), sampleUnits.threadCPUDelta: "variable CPU cycles",
+ *    max_{sample}(sample.cpuDelta / sample.timeDelta) == 1234567 cycles per ms
+ *    Returns 1234567 * 3, i.e. "3703701 cycles per sample if each sample ticks at
+ *    the declared 3ms interval and the CPU usage is at the observed maximum".
+ */
+export function computeMaxCPUDeltaPerInterval(profile: Profile): number | null {
+  const sampleUnits = profile.meta.sampleUnits;
+  if (!sampleUnits) {
+    return null;
+  }
+
+  const interval = profile.meta.interval;
+  const threadCPUDeltaUnit = sampleUnits.threadCPUDelta;
+
+  switch (threadCPUDeltaUnit) {
+    case 'µs':
+    case 'ns': {
+      const cpuDeltaTimeUnitMultiplier =
+        getCpuDeltaTimeUnitMultiplier(threadCPUDeltaUnit);
+      return cpuDeltaTimeUnitMultiplier / interval;
```

Comment:
Great catch! You're completely right. This should be `* interval`. I think I let myself get confused by this code in `computeMaxThreadCPUDelta`:

```js
      const intervalFactor = (time[i] - time[i - 1]) / interval;
      const currentCPUPerInterval = cpuDelta / intervalFactor;
```

But here we have `cpuDelta / intervalFactor` which is `cpuDelta / ((time[i] - time[i - 1]) / interval)` which is `(cpuDelta / (time[i] - time[i - 1]) * interval`.
So yes, `* interval` is correct. Fixed.

Reply:
I think it wasn't that visible in the test profiles 1) it's not used when we have cpu numbers 2) because activity graph still looks proportionate even with the wrong max value. Thanks for the fix!